impala/dbapi.py

```diff
@@ -43,7 +44,7 @@ def connect(host='localhost', port=21050, database=None, timeout=None,
             ldap_user=None, ldap_password=None, use_kerberos=None,
             protocol=None, krb_host=None, use_http_transport=False,
             http_path='', auth_cookie_names=['impala.auth', 'hive.server2.auth'],
-            retries=3):
+            retries=3, jwt=None):
```

Comment:
JWT tokens may also include expiration time in their payloads, so passing the JWT token directly to connect() may cause issues with long-running queries or when the same connection is used to execute several queries.

Maybe connect() could take a shell command instead that would return a fresh JWT token every time impyla makes an HTTP request?

(We do something similar for GSSAPI over HTTP, see transport.setGetCustomHeadersFunc() call in get_http_transport()).

Reply:
I think there are going to be deployments that use JWTs where the expiration time is measured in hours or days. In that case, it is useful to have a simple option of just passing in the JWT. In particular, I think some clients won't know how to get a fresh JWT, because they assume that the JWT will last for a day and they are never running that long.

I agree that we will want a way to handle expiration more gracefully. It makes sense that a user could provide a callback that provides a JWT. For now, I think having the easy way that just passes in a JWT in a good place to start. We can always add additional parameters to connect() to provide different ways to maintain the JWT.

Reply:
OK, thanks for the context. LGTM.